SmartDeviceLink-iOS/SmartDeviceLink/SDLProtocolHeader.m

```diff
@@ -38,20 +38,27 @@ - (NSString *)description {
     return description;
 }
 
-
 + (SDLProtocolHeader *)headerForVersion:(UInt8)version {
-    if (version == 1) {
-        return [[SDLV1ProtocolHeader alloc] init];
-    }
-
-    if (version == 2) {
-        return [[SDLV2ProtocolHeader alloc] init];
-    }
-
     // TODO: some error handling here if unknown version is asked for,
     // but that needs to be balanced against future proofing. i.e. V3.
     // Requirements around V3 are as yet undefined so give a V2 header
-    return [[SDLV2ProtocolHeader alloc] init];
+    // lol wat.
+    switch (version) {
+        case 0: {
+            return [[SDLV2ProtocolHeader alloc] initWithVersion:2];
+        } break;
+        case 1: {
+            return [[SDLV1ProtocolHeader alloc] init];
+        } break;
+        case 2:
+        case 3:
+        case 4: {
+            return [[SDLV2ProtocolHeader alloc] initWithVersion:version];
+        } break;
+        default: {
+            return [[SDLV2ProtocolHeader alloc] initWithVersion:2];
+        } break;
+    }
 }
```

Comment:
Because we are considering the protocol header version as a separate version. So it's a ProtocolHeader version 2, since in version 2, 3, 4, the header is all the same, but we need a specific version number for the protocol header itself. This is to avoid creating classes that mirror each other with hardcoded version values as the only difference.